enhancements/installer/aws-custom-edge-machineset-local-zones.md

```diff
@@ -0,0 +1,933 @@
+across the zones available to create the nodes (control plane and workers), querying the
+API [DescribeInstanceTypeOfferings](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeInstanceTypeOfferings.html).
+
+The [instance type offerings](https://aws.amazon.com/ec2/pricing/on-demand/) on
+Local Zones are different than the regular zones (see [exported instance offerings matrix on US Local Zones](https://gist.github.com/mtulio/c98aa15128a7becb06a372f00d824c42)).
+For that reason, the installer will lookup the offering for each zone local along
+the [preferred types](https://github.com/openshift/installer/blob/master/pkg/types/aws/defaults/platform.go#L38) when
+creating the MachineSets manifests for the `edge` pools. To cover all Local Zones
+location offerings, the type `c5d.2xlarge` will be appended to the preferred list.
+With that strategy, all the current GA locations will be covered with supported instance
+types, and when the most recent instances (currently 6th-gen `m6i.xlarge`) will be
+offered on those locations, it will be selected automatically without needing to
+change the installer.
+
+The logic of the preferred instance type will be replaced for the `edge` pool when
+the user provides the instance type on the install-config, attribute `compute[name=edge].platform.aws.type`.
+
+#### Volume type
+
+The EBS Volume Type `gp3`, used by default on IPI, is not broadly available
+across all GA Local Zones locations. [The Local Zones Features page](https://aws.amazon.com/about-aws/global-infrastructure/localzones/features) is the current place to check which location supports `gp3`.
+
+AWS also does provide an API to query the zone offering for EBS flavors. The EBS
+volume type for Local Zones, `edge` machine pool, will be fixed to `gp2`.
+
+The EBS Volume Type can be changed on the `edge` machine pool option
+`compute[name=edge].platform.aws.rootVolume.type` on the `install-config.yaml`.
+
+
+#### edge MachineSets
+
+The installer must generate one MachineSet for each Local Zones location, considering
+all the limitations on the Local Zones (instance types, EBS, network, etc).
+
+The MachineSet for the `edge` pool will also set those custom labels:
+
+- `node-role.kubernetes.io/edge=''`: used by `taint` and to quickly identify on `oc get nodes`
+- `location=local-zone`: can be used by customers to deploy custom applications across all Local Zones nodes
+- `zone_group=<zone_group_name>`: can be used by the developer to deploy applications in custom locations
+
+The MachineSet also must have the taint to `NoSchedule` to the label `node-role.kubernetes.io/edge=''`.
+This taint will prevent to schedule of regular cluster workloads (router, logging, monitoring)
+into distant geo-locations (Local Zones), allowing more control to the user to
+choose the workloads running for `edge` location, and also more control of the costs.
+
+#### General considerations
+
+The following items must be satisfied in this phase:
```

Comment:
I think we should also state that any zones specified in the edge pool `compute[?name=='edge'].platform.aws.zones` must have a corresponding subnet declared in `platform.aws.subnets`. This should also be true for phase 2.

That is, the following config would be invalid in both phase 1 & phase 2, because there is no supplied subnet for the zone specified in the `edge` pool (in phase 2, the installer will not create missing subnets when a user supplies existing subnets).

```yaml
apiVersion: v1
baseDomain: example.com
metadata:
  name: edge-cluster
compute:
- name: edge
  platform:
    aws:
      zones:
      - us-east-1-mia-1a
platform:
  aws:
    region: us-east-1
    subnets:
    - subnetID-us-east-1a
    - subnetID-us-east-1b
    - subnetID-us-east-1c
    - subnetID-us-east-1-nyc-1a
pullSecret: '{"auths": ...}'
sshKey: ssh-ed25519 AAAA...
```

Reply:
@patrickdillon thanks for pointing it out. I will make sure this EP and implementation covers it.